draft-ietf-tls-dtls-connection-id.md

```diff
@@ -134,44 +133,67 @@ indicates that the server will send with the client's CID but does not
 wish the client to include a CID (or again, alternately, to use a
 zero-length CID).
 
-When a session is resumed, the "connection_id" extension is
-negotiated afresh, not retained from previous connections in
-the session.
-
-Because each party sends the value in the "connection_id" extension that it wants to 
-receive as a connection identifier in encrypted records, it is possible
+Because each party sends the value in the "connection_id" extension it wants to 
+receive as a CID in encrypted records, it is possible
 for an endpoint to use a globally constant length for such connection
 identifiers.  This can in turn ease parsing and connection lookup,
-
 for example by having the length in question be a compile-time constant.
 Implementations, which want to use variable-length CIDs, are responsible
 for constructing the CID in such a way that its length can be determined
-on reception. Note that such implementations must still be able to send
-connection identifiers of different length to other parties.
-
-Note that it is not possible to parse the records without knowing how 
-long the CID is.
+on reception. Such implementations must still be able to send
+CIDs of different length to other parties. Note that there is no CID 
+length information included in the record itself.
 
 In DTLS 1.2, CIDs are exchanged at the beginning of the DTLS
 session only. There is no dedicated "CID update" message
 that allows new CIDs to be established mid-session, because
 DTLS 1.2 in general does not allow TLS 1.3-style post-handshake messages
-that do not themselves begin other handshakes. 
+that do not themselves begin other handshakes. When a DTLS session is 
+resumed or renegotiated, the "connection_id" extension is negotiated afresh. 
+
+If DTLS peers have not negotiated the use of a CIDs then the RFC 6347-defined 
+record format and content type MUST be used. 
+
+If DTLS peers have negotiated the use of a CIDs using the ClientHello and
+the ServerHello messages then the peers need to take the following steps.
 
-DTLS peers switch to the new record layer format, i.e., the record layer format 
-containing the CID, when encryption is enabled.
+The DTLS peers determine whether incoming and outgoing messages need 
+to use the new record format, i.e., the record format containing the CID. 
+The new record format and the CID content type is only used once encryption 
+is enabled. Plaintext payloads never use the new record type and the CID content 
+type. 
 
-# Record Layer Extensions and Record Payload Protection
+For sending, if a zero-length CID has been negotiated then the RFC 6347-defined 
```

Comment:
Why limit this with "for sending"?

Reply:
I wrote the text in a way that it talks about sending and receiving. This is just a writing style.